lib/jpegli/common_internal.h

```diff
@@ -83,16 +91,28 @@ constexpr uint32_t kJPEGZigZagOrder[64] = {
 template <typename T>
 class RowBuffer {
  public:
-  void Allocate(size_t num_rows, size_t stride) {
+  void Allocate(size_t num_rows, size_t rowsize) {
+    size_t vec_size = std::max(VectorSize(), sizeof(T));
```

Comment:
Why not hwy::kMaxVectorSize?
VectorSize depends on dynamic dispatch... looks dangerous.

Reply:
This is how it is done in ImageF and kMaxVectorSize is not recommended by hwy:
// kMaxVectorSize (undocumented, pending removal)

Reply:
kMaxVectorSize could be 2000+ on RVV/SVE...

Reply:
My concern is more about really dynamic dispatch.
On RVV/SVE vector size is known only in runtime but it is fixed.
But on x86, if implementation is swapped in the runtime (for whatever reason) vector size could change.
Likely we need a more sophisticated replacement of kMaxVectorSize in HWY, that would go through linked implementations and returns biggest vector size.